### ISSUE TITLE
[Bug Report] ...

### ISSUE THREAD

**Issue by bgoelTT** [tenstorrent/tt-metal#18983]
**Describe the bug**
When creating tensors from `torch.float32` huggingface-style weights in Llama-3.1-8B-Instruct, the PCC is much worse than if I were to first cast the weights as `torch.bfloat16`. This indicates there is a problem with the way we convert `torch.float32` tensors -> ttnn blockfpX tensors compared to converting `torch.bfloat16` tensors -> ttnn blockfpX tensors.

CC-ing @skhorasganiTT as point of contact for more detail.

**To Reproduce**
Unfortunately I cannot share the weights that produce this behaviour and do not have a great way of demonstrating the accuracy effects without also sharing the specific prompts I used. Can provide more detail outside of GH.

However, adding this code snippet below this line (https://github.com/tenstorrent/tt-metal/blob/2c77db965d415fd3a0f9addced9c214d7a76d14e/models/tt_transformers/tt/model_config.py#L1222) makes the model produce the correct responses:
```
for k, v in state_dict.items():
	state_dict[k] = v.to(torch.bfloat16)
```

**Expected behavior**
I expect we should be able to handle fp32 weights just as easily as we support bfloat16 weights.

**Screenshots**
N/A

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- TT-Metal commit `v0.56.0-rc45`

**Additional context**
Might need to take this offline for more detail.

**Comment by mtairum** [comment]
This is a great find. Have you opened an issue for our tensor conversor?

> We will probably need to check the weight PCCs for each submodule in 1 layer to find the problematic tensors.
Good idea. but this would only apply to float32 weights, right?

**Comment by skhorasganiTT** [comment]
Yes I meant for a fp32 checkpoint. You mean a separate issue from this one? No, the intention of this issue was to track it.

**Comment by mtairum** [comment]
Is this issue still relevant? 
CC: @gwangTT as it might be relevant for the wide TTT.

**Comment by gwangTT** [comment]
> Is this issue still relevant? CC: [@gwangTT](https://github.com/gwangTT) as it might be relevant for the wide TTT.

From TTT perspective, the code was updated to load bf16 weights if bf16 is the default dtype. But that may not answer the question about float32-to-blockfpX conversion.

**Comment by bgoelTT** [comment]
This applies if the weights are loaded into `torch.float32()` but it hasn't been tested in a while.

I still have the weights to share for repro
